app/assets/javascripts/application.js

```diff
@@ -42,42 +42,30 @@ function show_notification_blood(msg){
 }
 
 
-function prepare_to_initialize(q) {
+function prepare_to_initialize() {
   $(document).ready(function() {
-   google.maps.event.addDomListener(window, 'load',function () { initialize(q);});
-});
+   google.maps.event.addDomListener(window, 'load', initialize());
+ });
 }
 
 
- function initialize(x) {
+ function initialize() {
 
-  var input = document.getElementById(x);
+  var input = document.getElementsByClassName('search');
   var options = {
     componentRestrictions: {country: "eg"}
   };
+ var autocomplete = new google.maps.places.Autocomplete(input[0], options);
 
-  var autocomplete = new google.maps.places.Autocomplete(input,options);
-
- if (x == "searchTextField") {
-       google.maps.event.addListener(autocomplete, 'place_changed', function () {
+        google.maps.event.addListener(autocomplete, 'place_changed', function () {
         var place = autocomplete.getPlace();
-        document.getElementById('addName').value = document.getElementById('searchTextField').value;
-        document.getElementById('addLat').value = place.geometry.location.lat();
-        document.getElementById('addLng').value = place.geometry.location.lng();
-   });
-  }
-
-else {
-    google.maps.event.addListener(autocomplete, 'place_changed', function () {
-     var place = autocomplete.getPlace();
-     document.getElementById('hospital_addName').value = document.getElementById('hospital_searchTextField').value;
-     document.getElementById('hospital_addLat').value = place.geometry.location.lat();
-     document.getElementById('hospital_addLng').value = place.geometry.location.lng();
-   });
-  }
+        document.getElementsByClassName('add_name')[0].value = document.getElementsByClassName('search')[0].value;
+        document.getElementsByClassName('add_lat')[0].value = place.geometry.location.lat();
+        document.getElementsByClassName('add_lng')[0].value = place.geometry.location.lng();
+ });
```

Comment:
First: For google map is" outdated", every time I open PC and run app give
me these lines and I have to go the url and copy paste to the file
maps-google.js every time, I think this only for my PC but now I will
search to fix this problem.

Second: For naming class is there source for the rules of naming coz what I
find is rules related to something like camel case rules and so on.

Third: For indentation I use sublime,the problem is still although from the
first time I installed editorconfig added to package control of sublime.but
when add binding to "Key Bindings - User" file this fix the problem,
Thanks a lot.

I will modify the mentioned issues and commit again.



On Wed, Jul 27, 2016 at 6:19 PM, Ahmad Ajmi <notifications@github.com>
wrote:

> In app/assets/javascripts/application.js
> <https://github.com/Health-Care/BloodDonation/pull/18#discussion_r72471979>
> :
>
> > -        document.getElementById('addLng').value = place.geometry.location.lng();
> > -   });
> > -  }
> > -
> > -else {
> > -    google.maps.event.addListener(autocomplete, 'place_changed', function () {
> > -     var place = autocomplete.getPlace();
> > -     document.getElementById('hospital_addName').value = document.getElementById('hospital_searchTextField').value;
> > -     document.getElementById('hospital_addLat').value = place.geometry.location.lat();
> > -     document.getElementById('hospital_addLng').value = place.geometry.location.lng();
> > -   });
> > -  }
> > +        document.getElementsByClassName('add_name')[0].value = document.getElementsByClassName('search')[0].value;
> > +        document.getElementsByClassName('add_lat')[0].value = place.geometry.location.lat();
> > +        document.getElementsByClassName('add_lng')[0].value = place.geometry.location.lng();
> > + });
>
> Indentation please, you can install http://editorconfig.org/ and add this
> shortcut to sublime if you are using it
> https://coderwall.com/p/7yxpdw/auto-indenting-on-sublime-text-3
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Health-Care/BloodDonation/pull/18/files/a08fb7752af699997af7662f60c324cd1dc673a4..42358bd75297a9c725beb453a4d77ff4deb66a56#r72471979>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQ43pjpoTRzmAzHYyX4wgxPkulSlFKKaks5qZ4T5gaJpZM4IR2y0>
> .
>